src/lib/select/select.spec.ts

```diff
@@ -62,3203 +70,3284 @@ describe('MatSelect', () => {
   let scrolledSubject = new Subject();
   let viewportRuler: ViewportRuler;
 
-  beforeEach(fakeAsync(() => {
+  // Providers used for all mat-select tests
+  const commonProviders = [
+    {
+      provide: OverlayContainer, useFactory: () => {
+      overlayContainerElement = document.createElement('div') as HTMLElement;
+      overlayContainerElement.classList.add('cdk-overlay-container');
+
+      document.body.appendChild(overlayContainerElement);
+
+      // remove body padding to keep consistent cross-browser
+      document.body.style.padding = '0';
+      document.body.style.margin = '0';
+
+      return {getContainerElement: () => overlayContainerElement};
+    },
+    },
+    {provide: Directionality, useFactory: () => dir = {value: 'ltr'}},
+    {
+      provide: ScrollDispatcher, useFactory: () => ({
+      scrolled: () => scrolledSubject.asObservable(),
+    }),
+    },
+  ];
+
+  // NgModule imports used for all mat-select tests.
+  const commonModuleImports = [
+    MatFormFieldModule,
+    MatSelectModule,
+    ReactiveFormsModule,
+    FormsModule,
+    NoopAnimationsModule,
+  ];
+
+  /**
+   * Configures the test module for MatSelect with the given declarations. This is broken out so
+   * that we're only compiling the necessary test components for each test in order to speed up
+   * overall test time.
+   * @param declarations Components to declare for this block
+   */
+  function configureMatSelectTestingModule(declarations) {
```

Comment:
The declarations could at least be `any[]` to give it a little bit of type safety.

Reply:
I believe TypeScript infers the type here from how the argument is used